src/app/admin/components/DataTableSectionUser.tsx

```diff
@@ -0,0 +1,381 @@
+    setCurrentPage((prevPage) => Math.max(prevPage - 1, 1));
+  };
+
+  const handleNextPage = () => {
+    setCurrentPage((prevPage) => Math.min(prevPage + 1, totalPages));
+  };
+
+  const toggleEditMode = (rowId: string) => {
+    if (rowId === undefined) {
+      toast.error("❌ An error occurred", {
+        position: "top-right",
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        progress: 0,
+        theme: "light",
+        transition: Bounce,
+      });
+    } else if (rowId === editingId) {
+      setEditingId("");
+    } else {
+      setEditingId(rowId);
+    }
+    reset(); // reset form fields
+  };
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(e.target.value);
+    setCurrentPage(1);
+  };
+
+  const handleDeleteButton = async (userId: string) => {
+    setShowDeletePopup(true);
+    setRecordToDeleteId(userId);
+  };
+
+  return (
+    <div className="flex justify-center">
+      <div className={DATA_TABLE_SECTION_STYLES}>
+        {/* search results bar */}
+        <div className={SEARCH_RESULTS_SECTION_STYLES}>
+          <h1 className={SEARCH_RESULTS_TEXT_STYLES}>
+            Search Results (
+            {searchQuery === "" ? userData.length : filteredData.length}{" "}
+            {searchQuery === ""
+              ? userData.length === 1
+                ? "record"
+                : "records"
+              : "records"}{" "}
+            found)
+          </h1>
+          <input
+            type="text"
+            placeholder="Search id, name, email..."
+            className={SEARCH_BAR_STYLES}
+            onChange={handleSearchChange}
+          />
+
+          <Image
+            src={search_icon}
+            alt="Magnifying glass icon"
+            width={15}
+            height={15}
+            className={SEARCH_ICON_STYLES}
+          />
+        </div>
+        <div className={`${DATA_TABLE_CONTENT_STYLES} flex flex-col`}>
+          <div className="flex w-full flex-row text-white">
+            {tableHeaders.map((header, index) => (
+              <div
+                key={index}
+                className={`${DATA_TABLE_HEADER_CELL_STYLES} ${index === 0 ? "w-1/6" : index === 1 ? "w-1/6" : index === 2 ? "w-1/6" : index === 3 ? "w-1/6" : index === 4 ? "w-2/5" : "w-1/4"} border-r border-gray-300 `}
+              >
+                {header.columnHeader}
+              </div>
+            ))}
+          </div>
+
+          {currentPageData.map(
+            (rowData: Partial<Schema["User"]["type"]>, rowIndex: number) => (
+              <div
+                key={rowIndex}
+                className={`flex w-full flex-row ${rowIndex % 2 === 0 ? "bg-white" : "bg-light-grey"}`}
+              >
+                {editingId === rowData.id ? (
+                  <>
+                    <form
+                      onSubmit={handleSubmit((data) =>
+                        onSubmit(data, rowData.id as any),
```

Comment:
Looks weird, react hook form expects either an object or any type so i cant pass rowData.id as a string thus it is passed as any

Reply:
@burtonjong instead just get the id from the `editingId` in the handleSubmit method. No need to pass it in from here.